src/Tribe/Aggregator/Cron.php

```diff
@@ -385,6 +386,75 @@ public function verify_child_record_creation() {
 		}
 	}
 
+	public function start_batch_pushing_records() {
+		if ( ! tribe( 'events-aggregator.main' )->is_service_active() ) {
+			return;
+		}
+
+		$records = Tribe__Events__Aggregator__Records::instance();
+
+		$query = $records->query( [
+			'post_status'    => Tribe__Events__Aggregator__Records::$status->pending,
+			'posts_per_page' => 15,
+			'order'          => 'ASC',
+			'meta_query'     => [
+				'origin-not-csv'               => [
+					'key'     => '_tribe_aggregator_origin',
+					'value'   => 'csv',
+					'compare' => '!=',
+				],
+				'batch-push-support-specified' => [
+					'key'     => '_tribe_aggregator_allow_batch_push',
+					'value'   => true,
+					'compare' => '=',
+				],
+				'batch-not-queued'             => [
+					'key'     => '_tribe_aggregator_batch_started',
+					'value'   => 'bug #23268',
+					'compare' => 'NOT EXISTS',
+				],
+			],
+			'after'          => '-4 hours',
+		] );
+
+		if ( ! $query->have_posts() ) {
+			tribe( 'logger' )->log_debug( 'No Pending Batch to be started', 'EA Cron' );
+
+			return;
+		}
+
+		tribe( 'logger' )->log_debug( "Found {$query->found_posts} records", 'EA Cron' );
+
+		$cleaner = new Tribe__Events__Aggregator__Record__Queue_Cleaner();
+		foreach ( $query->posts as $post ) {
+			$record = $records->get_by_post_id( $post );
+
+			if ( $record === null || tribe_is_error( $record ) ) {
+				continue;
+			}
+
+			$cleaner
+				->set_stall_limit( HOUR_IN_SECONDS * 23 )
+				->set_time_to_live( HOUR_IN_SECONDS * 24 );
+
+			$cleaner->remove_duplicate_pending_records_for( $record );
+			$failed = $cleaner->maybe_fail_stalled_record( $record );
+
+			if ( $failed ) {
+				tribe( 'logger' )->log_debug( sprintf( 'Stalled record (%d) was skipped', $record->id ), 'EA Cron' );
+				continue;
+			}
+
+			// Just double Check for CSV
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 8cd4c09778a1840a8af5c89948e8bade</sub>